### PR TITLE
On pre response

### PR DESCRIPTION
**Pull request by orthagonal**

### CODE REVIEW

File: test/redirects.test.js [hunk 1]
```diff
@@ -404,4 +405,68 @@ lab.experiment('hapi-redirect', () => {
...
   });
+
+  lab.test(' accepts a callback that adds additional dynamic routes', (done) => {
+    let count = 0;
+    server.register({
+      register: redirectModule,
+      options: {
+        log: true,
+        log404: true,
+        getRedirects(slug, redirectDone) {
```
Comment (jgallen23) [inline test/redirects.test.js]: what is slug?

Comment (jgallen23) [reply test/redirects.test.js]: are we able to do both function redirects and static redirects at the same time?  if so, can we add a test for it?

File: test/redirects.test.js [hunk 2]
```diff
@@ -404,4 +405,68 @@ lab.experiment('hapi-redirect', () => {
...
+        }, (result2) => {
+          Code.expect(result2.statusCode).to.equal(302);
+          Code.expect(result2.headers.location).to.equal('/newtest2');
+          done();
+        });
+      });
+    });
+  });
+
+  lab.test('emits event when redirect occurs', (done) => {
```
Comment (jgallen23) [inline test/redirects.test.js]: can we do a quick scan of tests to make sure if there is a 404 and no redirect matches, that it just continues to send the 404

Comment (orthagonal) [reply test/redirects.test.js]: Added a test for that here: https://github.com/firstandthird/hapi-redirects/pull/28/files#diff-86f5351edab474a541ab1bcf9f7b5a6fR484

File: test/redirects.test.js [hunk 3]
```diff
@@ -404,4 +405,98 @@ lab.experiment('hapi-redirect', () => {
...
+    let count = 0;
+    server.register({
+      register: redirectModule,
+      options: {
+        log: true,
+        log404: true,
+        redirects: {
+          '/test301': '/it/works',
+        },
+        getRedirects(slug, redirectDone) {
```
Comment (jgallen23) [inline test/redirects.test.js]: What is slug?

File: test/redirects.test.js (hunk 1 above)
Comment (jgallen23) [reply test/redirects.test.js]: Ping

File: test/redirects.test.js (hunk 3 above)
Comment (orthagonal) [reply test/redirects.test.js]: Think 'slug' this was left over from the pagedata redirects.   I'm passing the plugin options to this callback now.

**Pull request merged**